File: typescript/src/methods/node.ts
import type { Client } from "../client.js";
import type { VersionResult } from "../types.js";

export async function getVersion(client: Client): Promise<string> {
  const result = await client.call<Record<string, never>, VersionResult>("getVersion", {});
  return result.version;
}

export async function getHealth(client: Client): Promise<string> {
  return client.call<Record<string, never>, string>("getHealth", {});
}

File: rust/src/methods/node.rs
//! Node methods (OFS-8200 §6's "Node" row).

use crate::client::Client;
use crate::error::Result;
use serde::Deserialize;

#[derive(Debug, Deserialize)]
struct VersionResult {
    version: String,
}

impl Client {
    pub async fn get_version(&self) -> Result<String> {
        Ok(self
            .call::<_, VersionResult>("getVersion", ())
            .await?
            .version)
    }

    pub async fn get_health(&self) -> Result<String> {
        self.call("getHealth", ()).await
    }
}
